src/Datasource/RuleInvoker.php

```diff
@@ -123,6 +123,9 @@ public function __invoke(EntityInterface $entity, array $scope): bool
         if (is_string($pass)) {
             $message = $pass;
         }
+        if (is_callable($message)) {
```

Comment:
Let's simply this check to `$message instanceof Closure`. Callable checks are a lot more expensive and just allowing a Closure would suffice most use cases.